src/chk/chk_engine.c

```diff
@@ -1394,26 +1483,49 @@ chk_engine_rejoin(void)
 	D_ASSERT(ins->ci_iv_group == NULL);
 	D_ASSERT(ins->ci_iv_ns == NULL);
 
-	need_join = true;
 	ins->ci_starting = 1;
 
-	rc = chk_prop_fetch(prop, &rank_list);
-	if (rc != 0)
-		goto out;
+	if (chk_is_on_leader(cbk->cb_gen, prop->cp_leader, true)) {
+		ins->ci_iv_ns = chk_leader_get_iv_ns();
+		/* If is possible that the check leader is not running. */
+		if (ins->ci_iv_ns == NULL)
+			goto out;
 
-	D_ASSERT(rank_list != NULL);
+		need_join = true;
+	} else {
+		need_join = true;
+
+		rc = chk_prop_fetch(prop, &rank_list);
+		if (rc != 0)
+			goto out;
 
-	rc = crt_group_secondary_create(CHK_DUMMY_POOL, NULL, rank_list, &ins->ci_iv_group);
+		D_ASSERT(rank_list != NULL);
+
+		rc = crt_group_secondary_create(CHK_DUMMY_POOL, NULL, rank_list, &ins->ci_iv_group);
+		if (rc != 0)
+			goto out;
+
+		uuid_parse(CHK_DUMMY_POOL, dummy_pool);
+		rc = ds_iv_ns_create(dss_get_module_info()->dmi_ctx, dummy_pool, ins->ci_iv_group,
+				     &ins->ci_iv_id, &ins->ci_iv_ns);
+		if (rc != 0)
+			goto out;
+
+		ds_iv_ns_update(ins->ci_iv_ns, prop->cp_leader);
+	}
```

Comment:
seems a bit duplicate with chk_engine_start()?

Reply:
For the IV namespace part, it can be shared between check engine and check leader.